tests/Unit/Analyzer/FileVisitorTest.php

```diff
@@ -419,4 +420,32 @@ public function getStatic(): self
 
         $this->assertCount(1, $violations);
     }
+
+    public function test_it_can_parse_enum(): void
+    {
+        $code = <<< 'EOF'
+<?php
+namespace Root\Cars;
+enum Enum
+{
+    case Hearts;
+    case Diamonds;
+    case Clubs;
+    case Spades;
+}
+EOF;
+
+        /** @var FileParser $fp */
+        $fp = FileParserFactory::createFileParser(TargetPhpVersion::create('8.1'));
+        $fp->parse($code, 'relativePathName');
+
+        $cd = $fp->getClassDescriptions();
+
+        $violations = new Violations();
+
+        $notHaveDependencyOutsideNamespace = new Implement('MyInterface');
+        $notHaveDependencyOutsideNamespace->evaluate($cd[0], $violations, 'we want to add this rule for our software');
```

Comment:
I am not sure I got the goal of this test

Reply:
It's a simple rule @micheleorselli to check if we are parsing or not the enum class.
Without the code inside the `FileVisitor` the enum class is not parsed and this test it's red.
We can try to find a different test to be more explicit. Any ideas?